### PR TITLE
Adding instructions on using ngrok to access your local dev environment. 

### PR DESCRIPTION
**Pull request by newtewt**
# Description

Added steps that detail creating, downloading, and registering your ngrok account to your computer. 
Then starting the webapp and running ngrok to expose the port you're running on. So you can access your local environment on another device. This is a work around for service workers requiring a https site. 

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by newtewt** [comment]
@ngaruko  can you also review this and ensure you're able to get this to work.

File: README.md [hunk 1]
```diff
@@ -142,6 +142,19 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
...
+
+1. Create an ngrok account at https://ngrok.com/ 
+2. Follow instructions on downloading and linking your computer to your ngrok account.
+3. Start the webapp. This can be via docker, grunt, debug, horti, etc....
+3. Run ngrok and forward it towards the port you are running the webapp on.
+  * EX: For running webapp in docker locally using the docker instructions above `$ ./ngrok https 443`. This will forward the traffic from your ngrok url on https to 443 on your local machine. </br>
+  * EX: For runnning via horti, debug, or grunt where the api starts on port 5988. `$ ./ngrok https 5988` This will forward the traffic from your ngrok url on https to 5988 on your local machine. 
+3. You can then enter the url into our android app or browser and connect to your local dev environment.                
+* Example output from ngrok: Forwarding https://1661304e.ngrok.io -> http://localhost:5988
```
Comment (kennsippell) [inline README.md]: Maybe clarify that it would be the `https://1661304e.ngrok.io` address that you put into the app?

File: README.md [hunk 2]
```diff
@@ -142,6 +142,19 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
 
 Navigate your browser to [`http://localhost:5988/medic/login`](http://localhost:5988/medic/login).
 
+### Working on local environment with real devices.
```
Comment (kennsippell) [inline README.md]: Trailing period is a bit awkward

File: README.md [hunk 3]
```diff
@@ -142,6 +142,19 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
 
 Navigate your browser to [`http://localhost:5988/medic/login`](http://localhost:5988/medic/login).
 
+### Working on local environment with real devices. 
+
+This process is relevant for viewing the Medic webapp on mobile devices when the api service is run on a developer machine running Webapp >v3.5.0. Webapp v3.5.0 relies on service workers, which require a valid HTTPS certificate to function. Follow these steps to make your developer build accessible from your android device at the trusted url created by ngrok.
+
+1. Create an ngrok account at https://ngrok.com/ 
+2. Follow instructions on downloading and linking your computer to your ngrok account.
```
Comment (kennsippell) [inline README.md]: You can use markdown's auto number by just having `1.` for all bullets.

File: README.md [hunk 4]
```diff
@@ -142,6 +142,19 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
...
+3. Start the webapp. This can be via docker, grunt, debug, horti, etc....
```
Comment (kennsippell) [inline README.md]: I think `debug` isn't an actual option?

File: README.md [hunk 5]
```diff
@@ -142,6 +142,19 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
...
+3. Run ngrok and forward it towards the port you are running the webapp on.
+  * EX: For running webapp in docker locally using the docker instructions above `$ ./ngrok https 443`. This will forward the traffic from your ngrok url on https to 443 on your local machine. </br>
+  * EX: For runnning via horti, debug, or grunt where the api starts on port 5988. `$ ./ngrok https 5988` This will forward the traffic from your ngrok url on https to 5988 on your local machine.
```
Comment (kennsippell) [inline README.md]: When I try `ngrok https 5988` i get the help page. Do you mean `ngrok http 5988` ?

Comment (newtewt) [reply README.md]: yeah that's my bad. I was writing from memory and not looking at the command I ran.

File: README.md (hunk 4 above)
Comment (newtewt) [reply README.md]: I was thinking that if I set vscode to run api in debug I could use ngrok to hit it. I can remove that if you feel it's still not valid.

Comment (kennsippell) [reply README.md]: I'd recommend either removing it or adding a section to explain it. Whichever you feel is better?

Comment (newtewt) [reply README.md]: I went ahead and removed it. If it comes up as a question we can add it back later.

File: README.md [hunk 6]
```diff
@@ -142,6 +142,20 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
...
+
+1. Create an ngrok account at https://ngrok.com/ 
+1. Follow instructions on downloading and linking your computer to your ngrok account.
+1. Start the webapp. This can be via docker, grunt, debug, horti, etc....
+1. Run ngrok and forward it towards the port you are running the webapp on.
+    * EX: For running webapp in docker locally using the docker instructions above `$ ./ngrok http 443`. This will forward the traffic from your ngrok url on https to 443 on your local machine. </br>
+    * EX: For runnning via horti, debug, or grunt where the api starts on port 5988. `$ ./ngrok http 5988` This will forward the traffic from your ngrok url on https to 5988 on your local machine.
+    * Example output from ngrok: Forwarding https://1661304e.ngrok.io -> http://localhost:5988 
+1. You can then enter the url(https://1661304e.ngrok.io) into our android app or browser and connect to your local dev environment.
```
Comment (kennsippell) [inline README.md]: Since this is on the main page. Maybe link to the android app repository here for "android app" ?

File: README.md [hunk 7]
```diff
@@ -142,6 +142,20 @@ Then run either `node ./server.js` from the api directory or `grunt dev-api` fro
...
+### Working on local environment with real devices 
+
+This process is relevant for viewing the Medic webapp on mobile devices when the api service is run on a developer machine running Webapp >v3.5.0. Webapp v3.5.0 relies on service workers, which require a valid HTTPS certificate to function. Follow these steps to make your developer build accessible from your android device at the trusted url created by ngrok.
+
+1. Create an ngrok account at https://ngrok.com/ 
+1. Follow instructions on downloading and linking your computer to your ngrok account.
+1. Start the webapp. This can be via docker, grunt, debug, horti, etc....
+1. Run ngrok and forward it towards the port you are running the webapp on.
+    * EX: For running webapp in docker locally using the docker instructions above `$ ./ngrok http 443`. This will forward the traffic from your ngrok url on https to 443 on your local machine. </br>
+    * EX: For runnning via horti, debug, or grunt where the api starts on port 5988. `$ ./ngrok http 5988` This will forward the traffic from your ngrok url on https to 5988 on your local machine.
```
Comment (kennsippell) [inline README.md]: runnning

File: README.md (hunk 6 above)
Comment (newtewt) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (newtewt) [reply README.md]: and done

**Comment by kennsippell** [comment]
This is the proposed fix for #5340.

**Comment by newtewt** [comment]
@ngaruko  that's a good point. I'm not sure if we have any documents about how to use the non-branded version of our app. @abbyad  do you know if we have docs on that? 

Also, I hope someday we could load that page dynamically so we could select from a list of test instances instead of seeing beta/alpha that don't exist anymore. While also keeping the custom option.

**Comment by newtewt** [comment]
I'm going to merge this and if we decide that we need more detail about the custom url we can add later.

**Comment by abbyad** [comment]
@newtewt I don't know of that documentation existing, so could be added somewhere in medic-docs.

**Pull request merged**